File: gherkin/java/src/main/java/gherkin/ast/Rule.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class Rule extends Node {
    private final String keyword;
    private final String name;
    private final String description;
    private final List<Node> children;

    public Rule(
            Location location,
            String keyword,
            String name,
            String description,
            List<Node> children) {
        super(location);
        this.keyword = keyword;
        this.name = name;
        this.description = description;
        this.children = Collections.unmodifiableList(children);
    }

    public List<Node> getChildren() {
        return children;
    }

    public String getKeyword() {
        return keyword;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }
}

File: gherkin/dotnet/Gherkin/Ast/Rule.cs
using System.Collections.Generic;

namespace Gherkin.Ast
{
    public class Rule : IHasLocation, IHasDescription, IHasChildren
    {
        public Location Location { get; private set; }
        public string Keyword { get; private set; }
        public string Name { get; private set; }
        public string Description { get; private set; }
        public IEnumerable<IHasLocation> Children { get; private set; }

        public Rule(Location location, string keyword, string name, string description, IHasLocation[] children)
        {
            Location = location;
            Keyword = keyword;
            Name = name;
            Description = description;
            Children = children;
        }
    }
}
